src/editor/map/context_manager.cpp

```diff
@@ -605,10 +620,20 @@ void context_manager::save_map_as_dialog()
 	int overwrite_res = 1;
 	do {
 		input_name = old_input_name;
-		int res = dialogs::show_file_chooser_dialog_save(gui_.video(), input_name, _("Save the Map As"), ".map");
-		if (res == 0) {
+
+		gui2::tfile_dialog dlg;
+
+		dlg.set_title(_("Save the Map As"))
+		   .set_ok_label(_("Save"))
+		   .set_save_mode(true)
+		   .set_path(input_name)
+		   .set_extension(".map");
+
+		if(dlg.show(gui_.video())) {
+			input_name = dlg.path();
+
 			if (filesystem::file_exists(input_name)) {
-				res = gui2::show_message(gui_.video(), "",
+				int res = gui2::show_message(gui_.video(), "",
 						_("The file already exists. Do you want to overwrite it?"), gui2::tmessage::yes_no_buttons);
```

Comment:
This and line 675 uses 'Do you want' while a prompt in the dialog uses 'Do you wish'.